api/handlers/listhandler.py

```diff
@@ -480,13 +480,69 @@ def get(self, cont_name, list_name, **kwargs):
 
         # Authenticated or ticketed download request
         else:
-            self.response.app_iter = open(filepath, 'rb')
-            self.response.headers['Content-Length'] = str(fileinfo['size']) # must be set after setting app_iter
-            if self.is_true('view'):
-                self.response.headers['Content-Type'] = str(fileinfo.get('mimetype', 'application/octet-stream'))
+            signed_url = files.get_signed_url(file_path, file_system, filename=filename)
+            if signed_url:
+                self.redirect(signed_url)
             else:
-                self.response.headers['Content-Type'] = 'application/octet-stream'
-                self.response.headers['Content-Disposition'] = 'attachment; filename="' + filename + '"'
+                range_header = self.request.headers.get('Range', '')
```

Comment:
If it's not too much effort, could you please hide these changes from the PR by tracking range-reads as it's target branch (at least until that's merged).